meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -300,7 +302,7 @@ public void unsubscribeChildChanges(String key, ChildChangeListener listener) {
 
   @Override
   public void unsubscribeConnectStateChanges(ConnectStateChangeListener listener) {
-
+    _zkClient.subscribeStateChanges(new StateChangeListenerAdapter(listener));
```

Comment:
Just noticed that we don't return any error or return code if unsubscribe fails. I see this comment "No-op if the listener is not subscribed to the key.". Does it mean that caller has to call unsubscribe again if it fails? and if that's the case then how that can be communicated to user?

Reply:
Runtime Exception will be thrown if ZkClient unsubscribe failed.